src/groupeddataframe/show.jl

```diff
@@ -44,7 +44,7 @@ function Base.show(io::IO, gd::GroupedDataFrame;
         h -= 2 # two lines are already used for header and gap between groups
 
         h1 = h2 = h # display heights available for first and last groups
-        if N > 1
+        if !allrows && N > 1
```

Comment:
@ronisbr - there is also an additional kwarg in PrettyTables.jl that governs number of rows to be printed in a table. Can you please remind it its name? We should check how passing it interacts with this PR.

Reply:
One for sure I now see is `display_size`.

Reply:
I should point out that change to line 47 doesn't change any behaviour: arguments like `allrows` get forwarded to `show` methods later on. (All `!allrows && …` does is save a few arithmetic operations in the case where we know that we don't need to calculate heights.) So if there are other `kwargs` that override `allrows`, then it might be better to always set the default `display_size` like before.

Reply:
Before I did not suggest to make a change to check `allrows`, but rather to check `get(io, :limit, false)`. However, looking at the code - maybe it is OK, as if `:limit` is `false` then display size will be ignored in `show` for groups. Just please confirm that we have a test for this.

Reply:
Oh, does `:limit` take precedence over `allrows`? I have added a single test for whether `allrows=true` works even when display size is small, but don't have tests where I set `:limit` directly.

Reply:
> does `:limit` take precedence over `allrows`?

It depends how you define *precedence*. `allrows` will override `:limit` if `allrows` is set to `true`.
However, you can have `allrows=false` and `:limit=false`. We should make sure that even if we pass custom `display_size` to `show` internally all works OK (I think it will, but let us make sure).

Reply:
> @ronisbr - there is also an additional kwarg in PrettyTables.jl that governs number of rows to be printed in a table. Can you please remind it its name? We should check how passing it interacts with this PR.

You can select the number of rows that will be printed if `crop` is set to `:both` or `:vertical`. In this case, the number of printed rows (including header, table dividers, etc) are selected based on the display size.